util/util.go

```diff
@@ -94,6 +96,50 @@ func GetServiceIDByName(c rest.FrontClient, name string) (akid.ServiceID, error)
 	return akid.ServiceID{}, errors.Errorf("cannot determine project ID for %s", name)
 }
 
+func VerifyServiceByServiceID(c rest.FrontClient, serviceID akid.ServiceID) error {
+	unexpectedErrMsg := "Something went wrong while starting the Agent. " +
+		"Please contact Postman support (observability-support@postman.com) with the error details"
+	failedToVerifyServiceErrMsg := "Failed to verify service for given serviceID: %s\n"
+
+	// Check if service is already verified and cached
+	if service, found := serviceIDCache.Get(akid.String(serviceID)); found {
```

Comment:
Non-blocking: Blarg, is the cast still necessary?  Should we have a typed cache in go-utils?

How about `serviceID.String()` instead? (and on line 138)

Reply:
Not sure; I thought to use `serviceID.String()` initially, but then noticed that we had used `akid.String()` in other places also. So, decided to proceed with this.

Reply:
`akid.String` is old. `String()` is preferable.